OpenSim/Analyses/OutputReporter.h

```diff
@@ -0,0 +1,109 @@
+// INCLUDES
+//=============================================================================
+#include <OpenSim/Simulation/Model/Analysis.h>
+#include <OpenSim/Common/Reporter.h>
+#include <OpenSim/Common/STOFileAdapter.h>
+#include "osimAnalysesDLL.h"
+
+#ifdef SWIG
+    #ifdef OSIMANALYSES_API
+        #undef OSIMANALYSES_API
+        #define OSIMANALYSES_API
+    #endif
+#endif
+//=============================================================================
+//=============================================================================
+namespace OpenSim { 
+
+/**
+ * The OutputReporter Analysis is a wrapper for a TableReporter. It generates
+ * and writes TimeSeriesTables of Output values according to the names listed
+ * as properties of the Analysis. OutputReporter enables the AnalyzeTool to
+ * report on Outputs via the Analysis interface but backed by a TableReporter.
+ * The OutputReporter currently only supports Outputs of type: double, Vec3
+ * and SpatialVec. The OutputReporter will automatically write multiple files-
+ * a file for each supported Output type: <results_file_name>.sto (as doubles),
+ * <results_file_name>Vec3.sto and <results_file_name>SpatialVec.sto.
+ *
+ * Note that the internal tables are reset at the beginning of a simulation or
+ * AnalyzeTool::run() and does not append results to previous tables.
+ */
+class OSIMANALYSES_API OutputReporter : public Analysis {
+OpenSim_DECLARE_CONCRETE_OBJECT(OutputReporter, Analysis);
+
+public:
+OpenSim_DECLARE_LIST_PROPERTY(output_paths, std::string,
+    "The names of Outputs to be reported. To select specific Component Outputs"
+    ", provide its path name. For example, 'slider/tx/value' is an Output for "
+    "the value of a Coordinate 'tx' belonging to the Joint 'slider'.");
+
+//=============================================================================
+// METHODS
+//=============================================================================
+    OutputReporter(Model *model = nullptr) : Analysis(model) {
+        constructProperty_output_paths();
+        setName("OutputReporter");
+    }
+    OutputReporter(const std::string& fileName) : 
+            Analysis(fileName, false) {
+        constructProperty_output_paths();
+        setName("OutputReporter");
+    }
+
+    virtual ~OutputReporter() = default;
+
+protected:
+    //--------------------------------------------------------------------------
+    // ANALYSIS INTERFACE
+    //--------------------------------------------------------------------------
+    int begin(SimTK::State& s) override;
+    int step(const SimTK::State& s, int setNumber) override;
+    int printResults(const std::string& baseName,
+        const std::string& dir = "",
+        double dT = -1.0,
+        const std::string& extension = ".sto") override;
+
+private:
+    // Keep a reference to the underlying TableReporter(s) for this Analysis
+    SimTK::ReferencePtr< TableReporter_<double> > _tableReporterDouble;
```

Comment:
With paths potentially Absolute, relative or relative to top model, which ones are we using/assuming, can't see that except in doxygen comment. If users will be checkign the syntax using XMLBrowser in GUI the syntax for specification should be very clear.

Reply:
Relative to the Model or absolute work. I added more comments to the property comment that appears in XML.